src/gennylib/include/gennylib/conventions.hpp

```diff
@@ -417,12 +473,116 @@ struct convert<genny::RateSpec> {
         auto timeUnitYaml = Load(strRepr.substr(spacePos + delimiter.size()));
         auto timeUnit = timeUnitYaml.as<genny::TimeSpec>();
 
-        rhs = genny::RateSpec(timeUnit, opCount);
+        rhs = genny::BaseRateSpec(timeUnit, opCount);
 
         return true;
     }
 };
 
+/**
+ * Convert between YAML and genny::PercentileRateSpec
+ *
+ * The YAML syntax accepts [genny::Integer]%
+ * The syntax is interpreted as a percentage of the max throughput.
+ */
+template <>
+struct convert<genny::PercentileRateSpec> {
+    static Node encode(const genny::PercentileRateSpec& rhs) {
+        std::stringstream msg;
+        msg << rhs.percent << "%";
+        return Node{msg.str()};
+    }
+
+    static bool decode(const Node& node, genny::PercentileRateSpec& rhs) {
+        if (node.IsSequence() || node.IsMap()) {
+            return false;
+        }
+        auto strRepr = node.as<std::string>();
+
+        // Use percent as the delimiter.
+        const std::string delimiter = "%";
+        auto percentPos = strRepr.find(delimiter);
+
+        if (percentPos == std::string::npos || percentPos != strRepr.size() - 1) {
+            std::stringstream msg;
+            msg << "Invalid value for PercentileRateSpec field, expected an integer followed by %."
+                   " Saw: "
+                << strRepr;
+            throw genny::InvalidConfigurationException(msg.str());
+        }
+
+        auto percentYaml = Load(strRepr.substr(0, percentPos));
+        auto percent = percentYaml.as<genny::IntegerSpec>();
+
+        if (percent.value > 100) {
+            std::stringstream msg;
+            msg << "Invalid value for PercentileRateSpec field, integer cannot be greater than 100."
+                   " Saw: "
+                << percent.value;
+            throw genny::InvalidConfigurationException(msg.str());
+        }
+
+        rhs = genny::PercentileRateSpec(percent);
+
+        return true;
+    }
+};
+
+/**
+ * Convert between YAML and genny::RateSpec
+ *
+ * The YAML syntax accepts either [genny::Integer] per [genny::Time]
+ * or [genny::Integer]%
+ *
+ * The syntax is interpreted as operations per unit of time or
+ * percentage of max throughput.
+ */
+template <>
+struct convert<genny::RateSpec> {
```

Comment:
💅 can you put the `convert<>`s after the classes? It's technically possible to not have the `convert` structs if there is a ctor that takes in a `genny::Node`. No need to do that now, but putting convert next to impl will make that change easier

Reply:
This one's actually kinda tricky; all the structs and classes are defined in the `genny` namespace while all the `convert<>`s are in the `YAML` namespace, so all the converts are together. I think it's a bit better to not have to open and close the namespaces for each one or to resolve the `genny` scope every time.